javascript/diff_match_patch_uncompressed.js

```diff
@@ -68,27 +68,16 @@ var DIFF_EQUAL = 0;
 
 /**
  * Class representing one diff tuple.
- * Attempts to look like a two-element array (which is what this used to be).
+ * Constructor returns an actual two-element array, to allow destructuring
+ * See https://github.com/JackuB/diff-match-patch/issues/14 for details
  * @param {number} op Operation, one of: DIFF_DELETE, DIFF_INSERT, DIFF_EQUAL.
  * @param {string} text Text to be deleted, inserted, or retained.
  * @constructor
  */
 diff_match_patch.Diff = function(op, text) {
-  this[0] = op;
-  this[1] = text;
+  return [op, text];
 };
 
-diff_match_patch.Diff.prototype.length = 2;
-
-/**
- * Emulate the output of a two-element array.
- * @return {string} Diff operation as a string.
- */
-diff_match_patch.Diff.prototype.toString = function() {
-  return this[0] + ',' + this[1];
-};
```

Comment:
as with the previous comment, we should leave these lines in so we don't break the identify of the `Diff` object.

Reply:
OK, so to be clear you want to keep the Diff object and make it iterable?

Reply:
I'm circling back to this PR and realised I shelved the work until this question was answered and then forgot to follow up. Is my statement above correct?

Reply:
Yeah sorry for not responding earlier. Since this is a change about build-time semantics I think we should avoid changing runtime semantics. That is, if someone was already using these functions we don't want to inadvertently break their code just because we were trying to make it easier to import the library.

Reply:
was it previously iterable? I think what happened before is that we were setting the properties `0` and `1` on the object and it was possible to read those. it's array-like, but only because those properties are defined. I think it would still failure in an iteration like `diff.forEach()` (maybe I'm wrong, should be easy to confirm).

the biggest change between `return [a, b]` and `this[0] = a; this[1] = b` is that we're taking something that _was_ a class (or more specifically, an object with a specific constructor `Diff`), and turning it into a semantically void array of two elements.

we can preserve this while making it iterable to destructure it. maybe it would all look clearer using the `class` syntax.

```js
class Diff {
	constructor( op, text ) {
		this[0] = op;
		this[1] = text;
	}

	*[Symbol.iterator]() {
		yield this[0];
		yield this[1];
	}
}
```

(this example may be entirely broken)

Reply:
Yes, the specific reason this needs to be iterable is for destructuring, e.g. 
```
var a = dmp.diff_main('abc', 'ab123c', false)
const [eq, str] = a[0]
```
Which doesn't work now because `a[0]` is a `Diff` object. 

Whether it's a breaking change depends on who you think the target audience is. NPM users are IMO likely using the code I'm submitting here which came from the existing (now-abandoned) NPM package I mentioned above.
https://www.npmjs.com/package/diff-match-patch

As far as I can see, the history of the code is:
* Google used an array for a long time
* In 2018 they [swapped it to this `Diff` object](https://github.com/google/diff-match-patch/commit/cd60d246aecf871367313c7cf6dc8814af32c5e3), which isn't iterable
* The developer of the now-abandoned NPM package logged an issue about it as it's a breaking change https://github.com/google/diff-match-patch/issues/39
* In the meantime, that developer [reversed the change and made it an array again for NPM](https://github.com/JackuB/diff-match-patch/commit/18711ba8abef83aba02d30fe20206fad6d5c5487) - at the time I think this was expected to be temporary.

It looks like your suggested fix matches the one mentioned in that ticket on the google repo:
https://github.com/google/diff-match-patch/issues/39#issuecomment-457262957

So let's just include that.